File: Kattis/Python/Basic-programming1.py
from itertools import count
import math

def main():
    n, t = input().split()
    n = int(n)
    t = int(t)
    
    a = input().split()
    a = [int(i) for i in a]

    if t == 1:

        print("7\n")
    
    elif t == 2:

        if a[0] > a[1]:
            print("Bigger\n")
        elif a[0] < a[1]:
            print("Smaller\n")
        else:
            print("Equal\n")

    elif t == 3:
        
        c = []
        for i in range(3):
            c.append(a[i])
        c.sort()

        print(c[1], end="\n")

    elif t == 4:

        ans = sum(a)
        print(ans, end="\n")

    elif t == 5:

        ans = 0
        for i in range(n):
            if a[i] % 2 == 0:
                ans += a[i]

        print(ans, end="\n")


    elif t == 6:

        for i in range(n):
            a[i] = a[i] % 26
            print(chr(97 + a[i]), end="")

        print("\n")

    elif t == 7:

        i = 0
        count = 0
        while count < 2 * n:
            i = a[i]
            count += 1
            if i >= n:
                print("Out\n")
                return 0
            elif i == n - 1:
                print("Done\n")
                return 0
        
        print("Cyclic\n")
    
    return 0


if __name__ == "__main__":
    main()

File: Kattis/C++/Basic-Programming1/Basic-Programming1.cpp
#include <bits/stdc++.h>


using namespace std;
typedef long long i64;

int main()
{
    i64 n,t;

    cin >> n >> t;

    vector<i64> v(n);

    for(auto& i : v)
    {
        cin >> i;
    }

    if(t == 1)
    {
        cout << 7 << endl;
    }

    else if(t == 2)
    {
        if(v[0] > v[1])
        {
            cout << "Bigger" << endl;
        }
        else if(v[0] == v[1])
        {
            cout << "Equal" << endl;
        }
        else
        {
            cout << "Smaller" << endl;
        }
    }

    else if(t == 3)
    {
        sort(v.begin(),v.begin() + 3);
        cout << v[1] << endl;
    }

    else if(t == 4)
    {
        i64 m = 0;
        for(auto& i : v)
        {
            m += i;
        }
        cout << m << endl;
    }

    else if(t == 5)
    {
        i64 m = 0;
        for(auto& i : v)
        {
            if(i % 2 == 0)
            {
                m += i;
            }
        }
        cout << m << endl;
    }

    else if(t == 6)
    {
        for(auto& i : v)
        {
            i = i % 26;
            cout << char('a' + i);
        }
        cout << endl;
    }

    else if(t == 7)
    {
        i64 z = 0, c = 0;
        while(z < 999999)
        {
            z++;
            c = v[c];
            if(c > n - 1)
            {
                cout << "Out" << endl;
                return 0;
            }
            else if(c == n - 1)
            {
                cout << "Done" << endl;
                return 0;
            }
        }
        cout << "Cyclic" << endl;
    }
    return 0;
}
